### PR TITLE
✨ [RUMF-438] add user action reference to the internal context

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
Logs and traces will automatically pick up and associate this to their sent data.

### CODE REVIEW

File: packages/rum/src/userActionCollection.ts [hunk 1]
```diff
@@ -74,11 +74,16 @@ let currentUserAction: { id: string; startTime: number } | undefined
...
-    return { id: currentUserAction.id }
+export function getUserActionReference(time?: number): UserActionReference | undefined {
+  if (!currentUserAction) {
+    return undefined
   }
-  return undefined
+
+  if (time !== undefined && time < currentUserAction.startTime) {
+    return undefined
+  }
```
Comment (bcaudan) [inline packages/rum/src/userActionCollection.ts:84]: what about grouping these checks?

Comment (BenoitZugmeyer) [reply packages/rum/src/userActionCollection.ts:84]: I did at first, but I think this makes the function more readable. WDYT?

Comment (bcaudan) [reply packages/rum/src/userActionCollection.ts:84]: IMO:
```ts
if (!currentUserAction || (time !== undefined && time < currentUserAction.startTime)) {
```
has the same effect, you need understand both parts separately.
But having two if statements following each other with the same body bugged me a little, it does not seem very natural.
if you think that we should make the condition clearer, maybe extracting a function for the time predicate could clarify the intent.

Comment (BenoitZugmeyer) [reply packages/rum/src/userActionCollection.ts:84]: All right :)

File: packages/rum/src/rum.ts [hunk 2]
```diff
@@ -223,6 +223,7 @@ export function startRum(
       return {
         application_id: applicationId,
         session_id: viewContext.sessionId,
+        user_action: getUserActionReference(),
```
Comment (bcaudan) [inline packages/rum/src/rum.ts:226]: On my PR (😆), I missed the `InternalContext` type update, what about enforcing it with:
```ts
getInternalContext: monitor((): InternalContext => {
```

**Pull request merged**